--- Minecraft.java
+++ Minecraft.java
@@ -2,4 +2,7 @@
 
 import java.awt.event.WindowListener;
+
+import net.buj.rml.Environment;
+import net.buj.rml.RosepadModLoader;
 import net.minecraft.src.GameWindowListener;
 import net.minecraft.src.MinecraftImpl;
@@ -83,5 +86,5 @@
 import net.minecraft.src.PlayerController;
 
-public abstract class Minecraft implements Runnable {
+public abstract class Minecraft implements Runnable, net.buj.rml.MinecraftImpl {
     public PlayerController playerController;
     private boolean fullscreen;
@@ -120,4 +123,5 @@
     public MouseHelper mouseHelper;
     public File mcDataDir;
+    public RosepadModLoader modLoader;
     public static long[] frameTimes;
     public static int numRecordedFrameTimes;
@@ -135,5 +139,27 @@
     long systemTime;
     private int joinPlayerCounter;
+
+    public static Minecraft INSTANCE;
     
+    public int[] getVersion() {
+    	return new int[] { 1, 1, 0, 6 };
+    }
+    
+    public String getVersionTag() {
+    	return "dev";
+    }
+    
+    public String getVersionString() {
+    	int[] version = getVersion();
+    	StringBuilder str = new StringBuilder();
+    	for (int i = 0; i < version.length; i++) {
+    		if (i != 0) str.append(".");
+    		str.append(version[i]);
+    	}
+    	String tag = getVersionTag();
+    	if (tag.length() > 0) str.append("-").append(tag);
+    	return str.toString();
+    }
+    
     static {
         Minecraft.frameTimes = new long[512];
@@ -178,4 +204,7 @@
         this.displayHeight = height;
         this.fullscreen = fullscreen;
+        this.modLoader = new RosepadModLoader();
+
+        INSTANCE = this;
     }
     
@@ -211,5 +240,5 @@
             Display.setDisplayMode(new DisplayMode(this.displayWidth, this.displayHeight));
         }
-        Display.setTitle("Minecraft Minecraft Alpha v1.1.0");
+        Display.setTitle(getVersionString());
         try {
             PixelFormat pixelformat = new PixelFormat();
@@ -222,5 +251,5 @@
                 Thread.sleep(1000L);
             }
-            catch (InterruptedException ex3) {}
+            catch (InterruptedException ignore) {}
             Display.create();
         }
@@ -233,5 +262,5 @@
         Keyboard.create();
         Mouse.create();
-        this.mouseHelper = new MouseHelper((Component)this.mcCanvas);
+        this.mouseHelper = new MouseHelper(this.mcCanvas);
         try {
             Controllers.create();
@@ -268,7 +297,12 @@
             (this.downloadResourcesThread = new ThreadDownloadResources(this.mcDataDir, this)).start();
         }
-        catch (Exception ex4) {}
+        catch (Exception ignored) {}
         this.checkGLError("Post startup");
         this.ingameGUI = new GuiIngame(this);
+        {
+            File file = new File(getMinecraftDir(), "mods");
+            if (!file.exists() && file.mkdir()) doNothing();
+            this.modLoader.load(Environment.CLIENT, file);
+        }
         if (this.serverName != null) {
             this.displayGuiScreen(new GuiConnecting(this, this.serverName, this.serverPort));
@@ -280,5 +314,5 @@
     
     private void loadScreen() throws LWJGLException {
-        final ScaledResolution scaledResolution = new ScaledResolution(this.displayWidth, this.displayHeight);
+        final ScaledResolution scaledResolution = new ScaledResolution(this.displayWidth, this.displayHeight, true);
         final int scaledWidth = scaledResolution.getScaledWidth();
         final int scaledHeight = scaledResolution.getScaledHeight();
@@ -304,6 +338,4 @@
         instance.addVertexWithUV(0.0, 0.0, 0.0, 0.0, 0.0);
         instance.draw();
-        final int integer5 = 256;
-        final int integer6 = 256;
         GL11.glColor4f(1.0f, 1.0f, 1.0f, 1.0f);
         instance.setColorOpaque_I(16777215);
@@ -317,6 +349,4 @@
     
     public void scaledTessellator(final int integer1, final int integer2, final int integer3, final int integer4, final int integer5, final int integer6) {
-        final float n = 0.00390625f;
-        final float n2 = 0.00390625f;
         final Tessellator instance = Tessellator.instance;
         instance.startDrawingQuads();
@@ -341,5 +371,5 @@
             case 1:
             case 2: {
-                file = new File(property, String.valueOf('.') + appDirName + '/');
+                file = new File(property, '.' + appDirName + '/');
                 break;
             }
@@ -347,12 +377,12 @@
                 final String getenv = System.getenv("APPDATA");
                 if (getenv != null) {
-                    file = new File(getenv, new StringBuilder(".").append(appDirName).append('/').toString());
+                    file = new File(getenv, "." + appDirName + '/');
                     break;
                 }
-                file = new File(property, String.valueOf('.') + appDirName + '/');
+                file = new File(property, '.' + appDirName + '/');
                 break;
             }
             case 4: {
-                file = new File(property, new StringBuilder("Library/Application Support/").append(appDirName).toString());
+                file = new File(property, "Library/Application Support/" + appDirName);
                 break;
             }
@@ -363,5 +393,5 @@
         }
         if (!file.exists() && !file.mkdirs()) {
-            throw new RuntimeException(new StringBuilder().append("The working directory could not be created: ").append(file).toString());
+            throw new RuntimeException("The working directory could not be created: " + file);
         }
         return file;
@@ -407,5 +437,5 @@
             this.setIngameNotInFocus();
             final ScaledResolution scaledResolution = new ScaledResolution(this.displayWidth, this.displayHeight);
-            guiScreen.setWorldAndResolution(this, scaledResolution.getScaledWidth(), scaledResolution.getScaledHeight());
+            guiScreen.setWorldAndResolution(this, scaledResolution);
             this.skipRenderWorld = false;
         }
@@ -420,6 +450,6 @@
             final String gluErrorString = GLU.gluErrorString(glGetError);
             System.out.println("########## GL ERROR ##########");
-            System.out.println(new StringBuilder("@ ").append(location).toString());
-            System.out.println(new StringBuilder().append(glGetError).append(": ").append(gluErrorString).toString());
+            System.out.println("@ " + location);
+            System.out.println(glGetError + ": " + gluErrorString);
             System.exit(0);
         }
@@ -435,5 +465,5 @@
             }
         }
-        catch (Exception ex) {}
+        catch (Exception ignored) {}
         try {
             System.out.println("Stopping!");
@@ -442,5 +472,5 @@
                 GLAllocation.deleteTexturesAndDisplayLists();
             }
-            catch (Exception ex2) {}
+            catch (Exception ignored) {}
             this.sndManager.closeMinecraft();
             Mouse.destroy();
@@ -498,5 +528,5 @@
                 GL11.glEnable(3553);
                 if (this.theWorld != null) {
-                    while (this.theWorld.updatingLighting()) {}
+                    while (this.theWorld.updatingLighting()) doNothing();
                 }
                 if (!this.skipRenderWorld) {
@@ -538,5 +568,5 @@
                 this.isGamePaused = (!this.isMultiplayerWorld() && this.currentScreen != null && this.currentScreen.doesGuiPauseGame());
                 while (System.currentTimeMillis() >= currentTimeMillis + 1000L) {
-                    this.debug = new StringBuilder().append(n).append(" fps, ").append(WorldRenderer.chunksUpdated).append(" chunk updates").toString();
+                    this.debug = (n > 100 ? "§a" : n >= 60 ? "§e" : "§c") + n + "§r fps, §b" + WorldRenderer.chunksUpdated + "§r chunk updates";
                     WorldRenderer.chunksUpdated = 0;
                     currentTimeMillis += 1000L;
@@ -545,5 +575,7 @@
             }
         }
-        catch (MinecraftError minecraftError) {}
+        catch (MinecraftError e) {
+            e.printStackTrace();
+        }
         catch (Throwable throwable2) {
             this.theWorld = null;
@@ -716,5 +748,5 @@
                 final int n = currentItem2.stackSize;
                 final ItemStack useItemRightClick = currentItem2.useItemRightClick(this.theWorld, this.thePlayer);
-                if (useItemRightClick != currentItem2 || (useItemRightClick != null && useItemRightClick.stackSize != n)) {
+                if (useItemRightClick != currentItem2 || useItemRightClick.stackSize != n) {
                     this.thePlayer.inventory.mainInventory[this.thePlayer.inventory.currentItem] = useItemRightClick;
                     this.entityRenderer.itemRenderer.resetEquippedProgress2();
@@ -770,5 +802,5 @@
                 this.resize(this.displayWidth, this.displayHeight);
             }
-            System.out.println(new StringBuilder().append("Size: ").append(this.displayWidth).append(", ").append(this.displayHeight).toString());
+            System.out.println("Size: " + this.displayWidth + ", " + this.displayHeight);
         }
         catch (Exception ex) {
@@ -788,5 +820,5 @@
         if (this.currentScreen != null) {
             final ScaledResolution scaledResolution = new ScaledResolution(width, height);
-            this.currentScreen.setWorldAndResolution(this, scaledResolution.getScaledWidth(), scaledResolution.getScaledHeight());
+            this.currentScreen.setWorldAndResolution(this, scaledResolution);
         }
     }
@@ -870,7 +902,4 @@
                 }
                 else {
-                    if (this.currentScreen == null) {
-                        continue;
-                    }
                     this.currentScreen.handleMouseInput();
                 }
@@ -905,4 +934,8 @@
                                 this.thePlayer.dropPlayerItemWithRandomChoice(this.thePlayer.inventory.decrStackSize(this.thePlayer.inventory.currentItem, 1), false);
                             }
+                            if (InputHandler.GetKBEventKey() == this.gameSettings.keyBindHideGUI.keyCode && this.ingameGUI != null
+                                    && this.currentScreen == null && this.thePlayer != null) {
+                                this.ingameGUI.showGUI = !this.ingameGUI.showGUI;
+                            }
                             if (this.isMultiplayerWorld() && InputHandler.GetKBEventKey() == this.gameSettings.keyBindChat.keyCode) {
                                 this.displayGuiScreen(new GuiChat());
@@ -1003,7 +1036,5 @@
             else if (this.thePlayer != null) {
                 this.thePlayer.preparePlayerToSpawn();
-                if (world != null) {
-                    world.spawnEntityInWorld(this.thePlayer);
-                }
+                world.spawnEntityInWorld(this.thePlayer);
             }
             if (!world.multiplayerWorld) {
@@ -1051,8 +1082,8 @@
                 this.loadingScreen.setLoadingProgress(n2++ * 100 / n3);
                 this.theWorld.getBlockID(spawnX + i, 64, spawnZ + j);
-                while (this.theWorld.updatingLighting()) {}
             }
         }
         this.loadingScreen.displayLoadingString("Simulating world for a bit");
+        while (this.theWorld.updatingLighting()) doNothing();
         n3 = 2000;
         this.theWorld.dropOldChunks();
@@ -1093,5 +1124,5 @@
     
     public String debugLoadedEntities() {
-        return new StringBuilder("P: ").append(this.effectRenderer.getStatistics()).append(". T: ").append(this.theWorld.getDebugLoadedEntities()).toString();
+        return "P: §5" + this.effectRenderer.getStatistics() + "§f. T: " + this.theWorld.getDebugLoadedEntities();
     }
     
@@ -1114,5 +1145,5 @@
     
     public static boolean isValidName(final String username) {
-        return username.matches("[a-zA-Z0-9_ \\u0107 \\u015b]*");
+        return username.matches("[a-zA-Z\\d_ \\u0107\\u015b]*");
     }
     
@@ -1124,19 +1155,19 @@
         final Frame frame = new Frame("Minecraft");
         final Canvas canvas = new Canvas();
-        frame.setLayout((LayoutManager)new BorderLayout());
-        frame.add((Component)canvas, "Center");
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas, "Center");
         canvas.setPreferredSize(new Dimension(854, 480));
         frame.pack();
-        frame.setLocationRelativeTo((Component)null);
-        final MinecraftImpl minecraft = new MinecraftImpl((Component)frame, canvas, null, 854, 480, false, frame);
-        final Thread thread = new Thread((Runnable)minecraft, "Minecraft main thread");
+        frame.setLocationRelativeTo(null);
+        final MinecraftImpl minecraft = new MinecraftImpl(frame, canvas, null, 854, 480, false, frame);
+        final Thread thread = new Thread(minecraft, "Minecraft main thread");
         thread.setPriority(10);
         minecraft.appletMode = false;
         minecraft.minecraftUri = "www.minecraft.net";
-        if (username != null && sessionID != null) {
+        if (sessionID != null) {
             minecraft.session = new Session(username, sessionID);
         }
         else {
-            minecraft.session = new Session(new StringBuilder().append("Player").append(System.currentTimeMillis() % 1000L).toString(), "");
+            minecraft.session = new Session("Player" + System.currentTimeMillis() % 1000L, "");
         }
         if (server != null) {
@@ -1145,10 +1176,10 @@
         }
         frame.setVisible(true);
-        frame.addWindowListener((WindowListener)new GameWindowListener(minecraft, thread));
+        frame.addWindowListener(new GameWindowListener(minecraft, thread));
         thread.start();
     }
     
     public static void main(final String[] args) {
-        String string = new StringBuilder().append("Player").append(System.currentTimeMillis() % 1000L).toString();
+        String string = "Player" + System.currentTimeMillis() % 1000L;
         if (args.length > 0) {
             string = args[0];
@@ -1160,3 +1191,8 @@
         startMainThread(string, sessionID);
     }
+
+    /**
+     * IDEA shut up please
+     */
+    private void doNothing() {}
 }
