--- Minecraft.java
+++ Minecraft.java
@@ -136,4 +136,24 @@
     private int joinPlayerCounter;
     
+    public static int[] getVersion() {
+    	return new int[] { 1, 1, 0, 5 };
+    }
+    
+    public static String getVersionTag() {
+    	return "dev";
+    }
+    
+    public static String getVersionString() {
+    	int[] version = getVersion();
+    	StringBuilder str = new StringBuilder();
+    	for (int i = 0; i < version.length; i++) {
+    		if (i != 0) str.append(".");
+    		str.append(version[i]);
+    	}
+    	String tag = getVersionTag();
+    	if (tag.length() > 0) str.append("-").append(tag);
+    	return str.toString();
+    }
+    
     static {
         Minecraft.frameTimes = new long[512];
@@ -211,5 +231,5 @@
             Display.setDisplayMode(new DisplayMode(this.displayWidth, this.displayHeight));
         }
-        Display.setTitle("Minecraft Minecraft Alpha v1.1.0");
+        Display.setTitle(getVersionString());
         try {
             PixelFormat pixelformat = new PixelFormat();
@@ -280,5 +300,5 @@
     
     private void loadScreen() throws LWJGLException {
-        final ScaledResolution scaledResolution = new ScaledResolution(this.displayWidth, this.displayHeight);
+        final ScaledResolution scaledResolution = new ScaledResolution(this.displayWidth, this.displayHeight, true);
         final int scaledWidth = scaledResolution.getScaledWidth();
         final int scaledHeight = scaledResolution.getScaledHeight();
@@ -304,6 +324,4 @@
         instance.addVertexWithUV(0.0, 0.0, 0.0, 0.0, 0.0);
         instance.draw();
-        final int integer5 = 256;
-        final int integer6 = 256;
         GL11.glColor4f(1.0f, 1.0f, 1.0f, 1.0f);
         instance.setColorOpaque_I(16777215);
@@ -317,6 +335,4 @@
     
     public void scaledTessellator(final int integer1, final int integer2, final int integer3, final int integer4, final int integer5, final int integer6) {
-        final float n = 0.00390625f;
-        final float n2 = 0.00390625f;
         final Tessellator instance = Tessellator.instance;
         instance.startDrawingQuads();
@@ -407,5 +423,5 @@
             this.setIngameNotInFocus();
             final ScaledResolution scaledResolution = new ScaledResolution(this.displayWidth, this.displayHeight);
-            guiScreen.setWorldAndResolution(this, scaledResolution.getScaledWidth(), scaledResolution.getScaledHeight());
+            guiScreen.setWorldAndResolution(this, scaledResolution);
             this.skipRenderWorld = false;
         }
@@ -538,5 +554,5 @@
                 this.isGamePaused = (!this.isMultiplayerWorld() && this.currentScreen != null && this.currentScreen.doesGuiPauseGame());
                 while (System.currentTimeMillis() >= currentTimeMillis + 1000L) {
-                    this.debug = new StringBuilder().append(n).append(" fps, ").append(WorldRenderer.chunksUpdated).append(" chunk updates").toString();
+                    this.debug = new StringBuilder(n > 100 ? "§a" : n >= 60 ? "§e" : "§c").append(n).append("§r fps, §b").append(WorldRenderer.chunksUpdated).append("§r chunk updates").toString();
                     WorldRenderer.chunksUpdated = 0;
                     currentTimeMillis += 1000L;
@@ -788,5 +804,5 @@
         if (this.currentScreen != null) {
             final ScaledResolution scaledResolution = new ScaledResolution(width, height);
-            this.currentScreen.setWorldAndResolution(this, scaledResolution.getScaledWidth(), scaledResolution.getScaledHeight());
+            this.currentScreen.setWorldAndResolution(this, scaledResolution);
         }
     }
@@ -905,4 +921,8 @@
                                 this.thePlayer.dropPlayerItemWithRandomChoice(this.thePlayer.inventory.decrStackSize(this.thePlayer.inventory.currentItem, 1), false);
                             }
+if (InputHandler.GetKBEventKey() == this.gameSettings.keyBindHideGUI.keyCode && this.ingameGUI != null
+&& this.currentScreen == null && this.thePlayer != null) {
+this.ingameGUI.showGUI = !this.ingameGUI.showGUI;
+}
                             if (this.isMultiplayerWorld() && InputHandler.GetKBEventKey() == this.gameSettings.keyBindChat.keyCode) {
                                 this.displayGuiScreen(new GuiChat());
@@ -1003,7 +1023,5 @@
             else if (this.thePlayer != null) {
                 this.thePlayer.preparePlayerToSpawn();
-                if (world != null) {
-                    world.spawnEntityInWorld(this.thePlayer);
-                }
+                world.spawnEntityInWorld(this.thePlayer);
             }
             if (!world.multiplayerWorld) {
@@ -1093,5 +1111,5 @@
     
     public String debugLoadedEntities() {
-        return new StringBuilder("P: ").append(this.effectRenderer.getStatistics()).append(". T: ").append(this.theWorld.getDebugLoadedEntities()).toString();
+        return new StringBuilder("P: §5").append(this.effectRenderer.getStatistics()).append("§f. T: ").append(this.theWorld.getDebugLoadedEntities()).toString();
     }
     
