--- AxisAlignedBB.java
+++ AxisAlignedBB.java
@@ -15,5 +15,5 @@
     
     static {
-        AxisAlignedBB.boundingBoxes = (List<AxisAlignedBB>)new ArrayList();
+        AxisAlignedBB.boundingBoxes = new ArrayList<>();
         AxisAlignedBB.numBoundingBoxesInUse = 0;
     }
@@ -31,5 +31,5 @@
             AxisAlignedBB.boundingBoxes.add(getBoundingBox(0.0, 0.0, 0.0, 0.0, 0.0, 0.0));
         }
-        return ((AxisAlignedBB)AxisAlignedBB.boundingBoxes.get(AxisAlignedBB.numBoundingBoxesInUse++)).setBounds(minX, minY, minZ, maxX, maxY, maxZ);
+        return AxisAlignedBB.boundingBoxes.get(AxisAlignedBB.numBoundingBoxesInUse++).setBounds(minX, minY, minZ, maxX, maxY, maxZ);
     }
     
@@ -156,5 +156,7 @@
     
     public boolean intersectsWith(final AxisAlignedBB aabb) {
-        return aabb.maxX > this.minX && aabb.minX < this.maxX && aabb.maxY > this.minY && aabb.minY < this.maxY && aabb.maxZ > this.minZ && aabb.minZ < this.maxZ;
+        return aabb.maxX > this.minX && aabb.minX < this.maxX &&
+                aabb.maxY > this.minY && aabb.minY < this.maxY &&
+                aabb.maxZ > this.minZ && aabb.minZ < this.maxZ;
     }
     
