--- BlockSponge.java
+++ BlockSponge.java
@@ -12,6 +12,8 @@
             for (int j = y - n; j <= y + n; ++j) {
                 for (int k = z - n; k <= z + n; ++k) {
-                    world.getBlockMaterial(i, j, k);
-                    final Material water = Material.WATER;
+if (world.getBlockMaterial(i, j, k) == Material.WATER) {
+world.setBlock(i, j, k, 0);
+world.notifyBlocksOfNeighborChange(i, j, k, 0);
+}
                 }
             }
