--- CanvasMinecraftApplet.java
+++ CanvasMinecraftApplet.java
@@ -6,14 +6,14 @@
 public class CanvasMinecraftApplet extends Canvas {
     final MinecraftApplet mcApplet;
-    
+
     public CanvasMinecraftApplet(final MinecraftApplet minecraftApplet) {
         this.mcApplet = minecraftApplet;
     }
-    
+
     public synchronized void addNotify() {
         super.addNotify();
         this.mcApplet.startMainThread();
     }
-    
+
     public synchronized void removeNotify() {
         this.mcApplet.shutdown();
