--- EntityRenderer.java
+++ EntityRenderer.java
@@ -316,5 +316,6 @@
                 GL11.glDisable(3008);
                 renderGlobal.drawBlockBreaking(thePlayer, this.mc.objectMouseOver, 0, thePlayer.inventory.getCurrentItem(), renderPartialTick);
-                renderGlobal.drawSelectionBox(thePlayer, this.mc.objectMouseOver, 0, thePlayer.inventory.getCurrentItem(), renderPartialTick);
+                if (this.mc.ingameGUI.showGUI)
+                    renderGlobal.drawSelectionBox(thePlayer, this.mc.objectMouseOver, 0, thePlayer.inventory.getCurrentItem(), renderPartialTick);
                 GL11.glEnable(3008);
             }
@@ -349,5 +350,6 @@
                 GL11.glDisable(3008);
                 renderGlobal.drawBlockBreaking(thePlayer, this.mc.objectMouseOver, 0, thePlayer.inventory.getCurrentItem(), renderPartialTick);
-                renderGlobal.drawSelectionBox(thePlayer, this.mc.objectMouseOver, 0, thePlayer.inventory.getCurrentItem(), renderPartialTick);
+                if (this.mc.ingameGUI.showGUI)
+                    renderGlobal.drawSelectionBox(thePlayer, this.mc.objectMouseOver, 0, thePlayer.inventory.getCurrentItem(), renderPartialTick);
                 GL11.glEnable(3008);
             }
@@ -362,5 +364,6 @@
             this.setupFog(1);
             GL11.glClear(256);
-            this.renderHand(renderPartialTick, i);
+            if (this.mc.ingameGUI.showGUI || this.mc.currentScreen != null)
+                this.renderHand(renderPartialTick, i);
             if (!this.mc.gameSettings.anaglyph) {
                 return;
