--- EntityPlayerSP.java
+++ EntityPlayerSP.java
@@ -23,5 +23,5 @@
         if (session != null && session.username != null && session.username.length() > 0) {
             this.skinUrl = "skin:///" + session.username;
-            System.out.println(new StringBuilder("Loading texture ").append(this.skinUrl).toString());
+            System.out.println("Loading texture " + this.skinUrl);
         }
         this.username = session.username;
@@ -63,5 +63,4 @@
             }
         }
-        final long milestone = this.worldObj.milestone;
         super.onLivingUpdate();
     }
@@ -83,5 +82,4 @@
         }
         this.heartsLife = this.heartsHalvesLife;
-        System.out.println(new StringBuilder().append("Damage taken: ").append(damage).toString());
         if (this.inventory.getTotalArmorValue() == 0) {
             this.health = 0;
