--- EntityLiving.java
+++ EntityLiving.java
@@ -292,5 +292,17 @@
     public boolean attackEntityFrom(final Entity entity, int damage) {
         if (this.worldObj.multiplayerWorld) {
-            damage = 0;
+            if (this.worldObj.rosepadContentEnabled() && (entity instanceof EntityPlayerSP || this instanceof EntityPlayerSP)) {
+                if (this.worldObj instanceof WorldClient) {
+                    ((WorldClient) this.worldObj).addToSendQueue(
+                            new Packet35EntityDamage().hit(
+                                    ((WorldClient) this.worldObj).getIDByEntity(this),
+                                    damage,
+                                    this.isDead
+                            )
+                    );
+                }
+                else damage = 0;
+            }
+            else damage = 0;
         }
         this.entityAge = 0;
@@ -326,4 +338,13 @@
         if (this.health <= 0) {
             this.worldObj.playSoundAtEntity(this, this.getDeathSound(), this.getSoundVolume(), (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2f + 1.0f);
+            if (this instanceof EntityPlayerSP && this.worldObj instanceof WorldClient) {
+                ((WorldClient) this.worldObj).addToSendQueue(
+                        new Packet35EntityDamage().hit(
+                                ((WorldClient) this.worldObj).getIDByEntity(this),
+                                damage,
+                                true
+                        )
+                );
+            }
             this.onDeath(entity);
         }
@@ -369,4 +390,5 @@
         }
         this.dead = true;
+        if (worldObj.multiplayerWorld) return;
         final int dropItemID = this.getDropItemID();
         if (dropItemID > 0) {
