--- GuiButton.java
+++ GuiButton.java
@@ -13,22 +13,22 @@
     public boolean enabled;
     public boolean visible;
-    
-    public GuiButton(final int integer1, final int integer2, final int integer3, final String message) {
-        this(integer1, integer2, integer3, 200, 20, message);
+
+    private Runnable onClick = null;
+
+    public GuiButton(final int id, final int posX, final int posY, final String text) {
+        this(id, posX, posY, 200, 20, text);
     }
-    
-    protected GuiButton(final int integer1, final int integer2, final int integer3, final int integer4, final int integer5, final String string) {
-        this.width = 200;
-        this.height = 20;
+
+    public GuiButton(final int id, final int posX, final int posY, final int width, final int height, final String text) {
         this.enabled = true;
         this.visible = true;
-        this.id = integer1;
-        this.xPosition = integer2;
-        this.yPosition = integer3;
-        this.width = integer4;
-        this.height = integer5;
-        this.displayString = string;
+        this.id = id;
+        this.xPosition = posX;
+        this.yPosition = posY;
+        this.width = width;
+        this.height = height;
+        this.displayString = text;
     }
-    
+
     public void drawButton(final Minecraft minecraft, final int mouseX, final int mouseY) {
         if (!this.visible) {
@@ -58,7 +58,22 @@
         }
     }
-    
-    public boolean mousePressed(final int integer1, final int integer2) {
-        return this.enabled && integer1 >= this.xPosition && integer2 >= this.yPosition && integer1 < this.xPosition + this.width && integer2 < this.yPosition + this.height;
+
+    public boolean isHovered(final int mouseX, final int mouseY) {
+        return this.enabled && mouseX >= this.xPosition && mouseY >= this.yPosition && mouseX < this.xPosition + this.width && mouseY < this.yPosition + this.height;
     }
+
+    public boolean listensForMouseEvents() {
+        return false;
+    }
+
+    public void mousePressed(final int mouseX, final int mouseY) {
+        if (onClick != null) onClick.run();
+    }
+    public void mouseMoved(final int mouseX, final int mouseY) {}
+    public void mouseReleased(final int mouseX, final int mouseY) {}
+
+    public GuiButton setOnClickListener(Runnable r) {
+        onClick = r;
+        return this;
+    }
 }
