--- GuiConnecting.java
+++ GuiConnecting.java
@@ -6,9 +6,10 @@
     public NetworkManager clientManager;
     public boolean cancelled;
-    
-    public GuiConnecting(final Minecraft minecraft, final String string, final int integer) {
+
+    public GuiConnecting(final Minecraft minecraft, final ConnectionChannelAdapter adapter) {
+        NetClientHandler.rosepadMeta = null; // Feel free to make an RP to remove this
         this.cancelled = false;
         minecraft.changeWorld1(null);
-        new ThreadConnectToServer(this, minecraft, string, integer).start();
+        new ThreadConnectToServer(this, minecraft, adapter).start();
     }
     
