--- GuiChat.java
+++ GuiChat.java
@@ -29,4 +29,14 @@
     @Override
     protected void keyTyped(final char character, final int key) {
+        if (character == '\u0016') {
+            String clipStr = getClipboardString();
+            if (clipStr == null) clipStr = "";
+            else clipStr = Util.filterWith(clipStr, " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_'abcdefghijklmnopqrstuvwxyz{|}~\u2302\u00c7\u00fc\u00e9\u00e2\u00e4\u00e0\u00e5\u00e7\u00ea\u00eb\u00e8\u00ef\u00ee\u00ec\u00c4\u00c5\u00c9\u00e6\u00c6\u00f4\u00f6\u00f2\u00fb\u00f9\u00ff\u00d6\u00dc\u00f8£\u00d8\u00d7\u0192\u00e1\u00ed\u00f3\u00fa\u00f1\u00d1ªº¿®¬½¼¡«»");
+            this.message += clipStr;
+            if (this.message.length() > 100) {
+                this.message = this.message.substring(0, 100);
+            }
+            return;
+        }
         if (key == 1) {
             this.mc.displayGuiScreen(null);
@@ -50,6 +60,6 @@
     @Override
     public void drawScreen(final int mouseX, final int mouseY, final float renderPartialTick) {
-        this.drawRect(2, this.height - 14, this.width - 2, this.height - 2, Integer.MIN_VALUE);
-        this.drawString(this.fontRenderer, new StringBuilder("> ").append(this.message).append((this.updateCounter / 6 % 2 == 0) ? "_" : "").toString(), 4, this.height - 12, 14737632);
+        this.drawRect(2, this.height - 14, this.width - 2, this.height - 2, 0x44000000);
+        this.drawString(this.fontRenderer, "> " + this.message + (this.updateCounter / 6 % 2 == 0 ? "_" : ""), 4, this.height - 12, 0xffe0e0e0);
     }
     
@@ -61,5 +71,4 @@
             }
             this.message = String.valueOf(this.message) + this.mc.ingameGUI.testMessage;
-            final int n = 100;
             if (this.message.length() > 100) {
                 this.message = this.message.substring(0, 100);
