--- GuiEditSign.java
+++ GuiEditSign.java
@@ -46,4 +46,14 @@
     @Override
     protected void keyTyped(final char character, final int key) {
+        if (character == '\u0016') {
+            String clipboardString = GuiScreen.getClipboardString();
+            if (clipboardString == null) {
+                clipboardString = "";
+            }
+            this.entitySign.signText[this.editLine] += clipboardString;
+            if (this.entitySign.signText[this.editLine].length() > 15) {
+            	this.entitySign.signText[this.editLine] = this.entitySign.signText[this.editLine].substring(0, 15);
+            }
+        }
         if (key == 200) {
             this.editLine = (this.editLine - 1 & 0x3);
@@ -65,4 +75,8 @@
     @Override
     public void drawScreen(final int mouseX, final int mouseY, final float renderPartialTick) {
+        if (this.entitySign == null) {
+            this.mc.displayGuiScreen(null);
+            return;
+        }
         this.drawDefaultBackground();
         this.drawCenteredString(this.fontRenderer, this.screenTitle, this.width / 2, 40, 16777215);
