--- GuiEditSign.java
+++ GuiEditSign.java
@@ -9,5 +9,5 @@
     private int updateCounter;
     private int editLine;
-    
+
     public GuiEditSign(final TileEntitySign tileEntitySign) {
         this.screenTitle = "Edit sign message:";
@@ -15,5 +15,5 @@
         this.entitySign = tileEntitySign;
     }
-    
+
     @Override
     public void actionPerformed() {
@@ -22,15 +22,15 @@
         this.controlList.add(new GuiButton(0, this.width / 2 - 100, this.height / 4 + 120, "Done"));
     }
-    
+
     @Override
     public void onGuiClosed() {
         Keyboard.enableRepeatEvents(false);
     }
-    
+
     @Override
     public void updateScreen() {
         ++this.updateCounter;
     }
-    
+
     @Override
     protected void actionPerformed(final GuiButton button) {
@@ -43,26 +43,59 @@
         }
     }
-    
+
     @Override
     protected void keyTyped(final char character, final int key) {
-        if (key == 200) {
-            this.editLine = (this.editLine - 1 & 0x3);
+        if (character == '\u0016') {
+            String clipboardString = GuiScreen.getClipboardString();
+            if (clipboardString == null) {
+                clipboardString = "";
+            }
+            else clipboardString = Util.filterWith(clipboardString, " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_'abcdefghijklmnopqrstuvwxyz{|}~\u2302\u00c7\u00fc\u00e9\u00e2\u00e4\u00e0\u00e5\u00e7\u00ea\u00eb\u00e8\u00ef\u00ee\u00ec\u00c4\u00c5\u00c9\u00e6\u00c6\u00f4\u00f6\u00f2\u00fb\u00f9\u00ff\u00d6\u00dc\u00f8£\u00d8\u00d7\u0192\u00e1\u00ed\u00f3\u00fa\u00f1\u00d1ªº¿®¬½¼¡«»");
+            while (this.editLine < 4) {
+                this.entitySign.signText[this.editLine] += clipboardString;
+                if (this.entitySign.signText[this.editLine].length() > 15) {
+                    if (this.editLine < 3) {
+                        String overflow = this.entitySign.signText[this.editLine++].substring(15);
+                        this.entitySign.signText[this.editLine] = this.entitySign.signText[this.editLine] + overflow;
+                        this.editLine--;
+                    }
+                    this.entitySign.signText[this.editLine] = this.entitySign.signText[this.editLine].substring(0, 15);
+                    this.editLine++;
+                }
+                else break;
+            }
         }
-        if (key == 208 || key == 28) {
-            this.editLine = (this.editLine + 1 & 0x3);
+        else if (key == 200) {
+            this.editLine = (this.editLine - 1) % 4;
         }
-        if (key == 14 && this.entitySign.signText[this.editLine].length() > 0) {
-            this.entitySign.signText[this.editLine] = this.entitySign.signText[this.editLine].substring(0, this.entitySign.signText[this.editLine].length() - 1);
+        else if (key == 208 || key == 28) {
+            this.editLine = (this.editLine + 1) % 4;
         }
-        if (" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_'abcdefghijklmnopqrstuvwxyz{|}~\u2302\u00c7\u00fc\u00e9\u00e2\u00e4\u00e0\u00e5\u00e7\u00ea\u00eb\u00e8\u00ef\u00ee\u00ec\u00c4\u00c5\u00c9\u00e6\u00c6\u00f4\u00f6\u00f2\u00fb\u00f9\u00ff\u00d6\u00dc\u00f8£\u00d8\u00d7\u0192\u00e1\u00ed\u00f3\u00fa\u00f1\u00d1ªº¿®¬½¼¡«»".indexOf((int)character) >= 0 && this.entitySign.signText[this.editLine].length() < 15) {
-            final StringBuilder sb = new StringBuilder();
-            final String[] signText = this.entitySign.signText;
-            final int editLine = this.editLine;
-            signText[editLine] = sb.append(signText[editLine]).append(character).toString();
+        else if (key == 14) {
+            if (this.entitySign.signText[this.editLine].length() > 0)
+                this.entitySign.signText[this.editLine] = this.entitySign.signText[this.editLine].substring(0, this.entitySign.signText[this.editLine].length() - 1);
+            else if (this.editLine > 0) {
+                this.editLine--;
+                if (this.entitySign.signText[this.editLine].length() > 0)
+                    this.entitySign.signText[this.editLine] = this.entitySign.signText[this.editLine].substring(0, this.entitySign.signText[this.editLine].length() - 1);
+            }
         }
+        else if (" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_'abcdefghijklmnopqrstuvwxyz{|}~\u2302\u00c7\u00fc\u00e9\u00e2\u00e4\u00e0\u00e5\u00e7\u00ea\u00eb\u00e8\u00ef\u00ee\u00ec\u00c4\u00c5\u00c9\u00e6\u00c6\u00f4\u00f6\u00f2\u00fb\u00f9\u00ff\u00d6\u00dc\u00f8£\u00d8\u00d7\u0192\u00e1\u00ed\u00f3\u00fa\u00f1\u00d1ªº¿®¬½¼¡«»".indexOf((int)character) >= 0) {
+            if (this.entitySign.signText[this.editLine].length() < 15) {
+                this.entitySign.signText[this.editLine] += character;
+            }
+            else if (this.editLine < 3) {
+                this.editLine++;
+                this.entitySign.signText[this.editLine] += character;
+            }
+        }
     }
-    
+
     @Override
     public void drawScreen(final int mouseX, final int mouseY, final float renderPartialTick) {
+        if (this.entitySign == null) {
+            this.mc.displayGuiScreen(null);
+            return;
+        }
         this.drawDefaultBackground();
         this.drawCenteredString(this.fontRenderer, this.screenTitle, this.width / 2, 40, 16777215);
