--- GuiMainMenu.java
+++ GuiMainMenu.java
@@ -83,7 +83,8 @@
         this.controlList.add(new GuiButton(1, this.width / 2 - 100, this.height / 4 + 48, "Singleplayer"));
         this.controlList.add(new GuiButton(2, this.width / 2 - 100, this.height / 4 + 72, "Multiplayer"));
-        this.controlList.add(new GuiButton(0, this.width / 2 - 1, this.height / 4 + 96 + 0 + 0, 98, 20, "Options..."));
-        this.controlList.add(new GuiButton(3, this.width / 2 - 100, this.height / 4 + 96 + 0 + 0, 98, 20, "Mods Menu"));
-        this.controlList.add(new GuiButton(4, this.width / 2 - 100, this.height / 4 + 120, "Quit Game"));
+        this.controlList.add(new GuiButton(0, this.width / 2 - -2, this.height / 4 + 96 + 0 + 0, 98, 20, "Options..."));
+        this.controlList.add(new GuiButton(4, this.width / 2 - 100, this.height / 4 + 96 + 0 + 0, 98, 20, "Mods Menu"));
+        this.controlList.add(new GuiButton(3, this.width / 2 - 100, this.height / 4 + 120, "Quit Game"));
+        ((GuiButton)this.controlList.get(3)).enabled = false;
         if (this.mc.session == null) {
             ((GuiButton)this.controlList.get(1)).enabled = false;
