--- GuiName.java
+++ GuiName.java
@@ -12,5 +12,5 @@
     public GuiName(final GuiScreen screen) {
         this.updateCounter = 0;
-        this.name = "";
+        this.name = GuiScreenInputPass.name;
         this.parentScreen = screen;
     }
@@ -41,4 +41,5 @@
         if (button.id == 0) {
             GuiScreenInputPass.name = this.name;
+            this.mc.displayGuiScreen(this.parentScreen);
         }
     }
