--- GuiScreen.java
+++ GuiScreen.java
@@ -9,4 +9,9 @@
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+
+import net.buj.rml.Game;
+import net.buj.rml.annotations.Nullable;
+import net.buj.rml.registry.Items;
 import net.minecraft.client.Minecraft;
 
@@ -15,4 +20,5 @@
     public int width;
     public int height;
+    public ScaledResolution size;
     protected List<GuiButton> controlList;
     public boolean allowUserInput;
@@ -23,5 +29,4 @@
     static {
         GuiScreen.currentID = "";
-        GuiScreen.currentID = "";
     }
     
@@ -32,5 +37,5 @@
     public GuiScreen() {
         this.hasInputHandler = false;
-        this.controlList = (List<GuiButton>)new ArrayList();
+        this.controlList = new ArrayList<>();
         this.allowUserInput = false;
         this.hasInputHandler = this.InputHandlerLoaded();
@@ -39,5 +44,5 @@
     public void drawScreen(final int mouseX, final int mouseY, final float renderPartialTick) {
         for (int i = 0; i < this.controlList.size(); ++i) {
-            ((GuiButton)this.controlList.get(i)).drawButton(this.mc, mouseX, mouseY);
+        	this.controlList.get(i).drawButton(this.mc, mouseX, mouseY);
         }
     }
@@ -64,5 +69,5 @@
         if (mouseButton == 0) {
             for (int i = 0; i < this.controlList.size(); ++i) {
-                final GuiButton button = (GuiButton)this.controlList.get(i);
+                final GuiButton button = this.controlList.get(i);
                 if (button.mousePressed(x, y)) {
                     this.mc.sndManager.playSoundFX("random.click", 1.0f, 1.0f);
@@ -79,5 +84,5 @@
     }
     
-    public void setWorldAndResolution(final Minecraft minecraft, final int width, final int height) {
+    public void setWorldAndResolution(final Minecraft minecraft, final ScaledResolution size) {
         this.mc = minecraft;
         if (this.hasInputHandler) {
@@ -85,6 +90,7 @@
         }
         this.fontRenderer = minecraft.fontRenderer;
-        this.width = width;
-        this.height = height;
+        this.width = size.getScaledWidth();
+        this.height = size.getScaledHeight();
+        this.size = size;
         this.actionPerformed();
     }
@@ -120,44 +126,44 @@
     public void handleKeyboardInput() {
         if (Keyboard.getEventKeyState()) {
-            if (this.hasInputHandler && InputHandler.cheatsEnabled) {
+            if (this.hasInputHandler && InputHandler.cheatsEnabled && this.mc.gameSettings.cheatMenuMode != 4) {
                 if (Keyboard.getEventKey() == 79) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '1';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 80) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '2';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 81) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '3';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 75) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '4';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 76) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '5';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 77) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '6';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 71) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '7';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 72) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '8';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 73) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '9';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 82) {
                     GuiScreen.currentID = String.valueOf(GuiScreen.currentID) + '0';
-                    System.out.println(new StringBuilder("Current entered ID: ").append(GuiScreen.currentID).toString());
+                    System.out.println("Current entered ID: " + GuiScreen.currentID);
                 }
                 else if (Keyboard.getEventKey() == 74) {
@@ -166,15 +172,33 @@
                 else if (Keyboard.getEventKey() == 78) {
                     try {
+                        Optional<Items.Item> item = Game.items.byId(Integer.parseInt(currentID));
+                        /*
                         if (Block.BLOCKS_LIST.length > Integer.parseInt(GuiScreen.currentID) && Block.BLOCKS_LIST[Integer.parseInt(GuiScreen.currentID)] != null) {
-                            this.mc.thePlayer.dropPlayerItemWithRandomChoice(new ItemStack(Block.BLOCKS_LIST[Integer.parseInt(GuiScreen.currentID)], 64), true);
+                            this.mc.thePlayer.dropPlayerItemWithRandomChoice(
+                                    new ItemStack(Block.BLOCKS_LIST[Integer.parseInt(GuiScreen.currentID)], mc.gameSettings.getCheatMenuItemAmount(0)), true);
                             System.out.println("Given block to the player");
                         }
-                        else if (Item.ITEMS_LIST[Integer.parseInt(GuiScreen.currentID)] != null) {
-                            this.mc.thePlayer.dropPlayerItemWithRandomChoice(new ItemStack(Item.ITEMS_LIST[Integer.parseInt(GuiScreen.currentID)], 1), true);
+                        else if (Item.ITEMS_LIST[Integer.parseInt(GuiScreen.currentID)] != null && (
+                                Item.ITEMS_LIST[Integer.parseInt(GuiScreen.currentID)].getVanilla() || mc.theWorld.rosepadContentEnabled())) {
+                            this.mc.thePlayer.dropPlayerItemWithRandomChoice(new ItemStack(Item.ITEMS_LIST[Integer.parseInt(GuiScreen.currentID)],
+                                    mc.gameSettings.getCheatMenuItemAmount(1)), true);
                             System.out.println("Given item to the player");
                         }
                         else {
-                            System.out.println(new StringBuilder("No block or item with ID ").append(GuiScreen.currentID).toString());
+                            System.out.println("No block or item with ID " + GuiScreen.currentID);
+                        }*/
+                        if (item.isEmpty()) {
+                            System.out.println("No item or block with ID " + GuiScreen.currentID);
                         }
+                        else {
+                            Items.Item item$2 = item.get();
+
+                            if (!item$2.isVanilla() && !mc.theWorld.rosepadContentEnabled()) {
+                                System.out.println("Cannot get a non-vanilla item on a vanilla server");
+                            }
+                            else {
+                                System.out.println("Given item to player");
+                            }
+                        }
                     }
                     catch (Exception ex) {
