--- InventoryPlayer.java
+++ InventoryPlayer.java
@@ -95,7 +95,7 @@
         }
         integer2 -= n2;
-        final ItemStack itemStack2;
-        final ItemStack itemStack = itemStack2 = this.mainInventory[n];
-        itemStack2.stackSize += n2;
+        final ItemStack itemStack = this.mainInventory[n];
+        itemStack.stackSize += n2;
+        if (itemStack.stackSize < 0) itemStack.stackSize = -127;
         this.mainInventory[n].animationsToGo = 5;
         return integer2;
@@ -263,10 +263,10 @@
     }
     
-    public boolean canHarvestBlock(final Block block) {
+    public boolean canHarvestBlock(final Block block, final World world) {
         if (block.material != Material.ROCK && block.material != Material.IRON && block.material != Material.CRAFTED_SNOW && block.material != Material.SNOW) {
             return true;
         }
         final ItemStack stackInSlot = this.getStackInSlot(this.currentItem);
-        return stackInSlot != null && stackInSlot.canHarvestBlock(block);
+        return stackInSlot != null && stackInSlot.canHarvestBlock(block, world);
     }
     
