--- Item.java
+++ Item.java
@@ -4,7 +4,8 @@
 
 public class Item {
-    protected static Random rand;
-    public static Item[] ITEMS_LIST;
-    public static Item SHOVEL;
+    protected static Random rand = new Random();
+
+    public static Item[] ITEMS_LIST = new Item[32000];
+    public static Item SHOVEL_IRON;
     public static Item PICKAXE_IRON;
     public static Item AXE_IRON;
@@ -98,9 +99,6 @@
     public static Item RECORD_HIDDEN_DEN;
     public static Item RECORD_LEMURIA;
-    public final int shiftedIndex;
-    protected int maxStackSize;
-    protected int maxDamage;
-    protected int iconIndex;
-    protected boolean bFull3D;
+    public static Item RECORD_43343138;
+    public static Item RECORD_ESTONIAN_RELIQUARY;
     public static Item obsidianArmor0;
     public static Item obsidianArmor1;
@@ -120,118 +118,159 @@
     public static Item edibleFire;
     public static Item flameberge;
-    
-    static {
-        Item.rand = new Random();
-        Item.ITEMS_LIST = new Item[32000];
-        Item.SHOVEL = new ItemShovel(0, 2).setIconIndex(82);
-        Item.PICKAXE_IRON = new ItemPickaxe(1, 2).setIconIndex(98);
-        Item.AXE_IRON = new ItemAxe(2, 2).setIconIndex(114);
-        Item.FLINT_AND_STEEL = new ItemFlintAndSteel(3).setIconIndex(5);
-        Item.APPLE = new ItemFood(4, 4).setIconIndex(10);
-        Item.BOW = new ItemBow(5).setIconIndex(21);
-        Item.ARROW = new Item(6).setIconIndex(37);
-        Item.COAL = new Item(7).setIconIndex(7);
-        Item.DIAMOND = new Item(8).setIconIndex(55);
-        Item.INGOT_IRON = new Item(9).setIconIndex(23);
-        Item.INGOT_GOLD = new Item(10).setIconIndex(39);
-        Item.SWORD_IRON = new ItemSword(11, 2).setIconIndex(66);
-        Item.SWORD_WOOD = new ItemSword(12, 0).setIconIndex(64);
-        Item.SHOVEL_WOOD = new ItemShovel(13, 0).setIconIndex(80);
-        Item.PICKAXE_WOOD = new ItemPickaxe(14, 0).setIconIndex(96);
-        Item.AXE_WOOD = new ItemAxe(15, 0).setIconIndex(112);
-        Item.SWORD_STONE = new ItemSword(16, 1).setIconIndex(65);
-        Item.SHOVEL_STONE = new ItemShovel(17, 1).setIconIndex(81);
-        Item.PICKAXE_STONE = new ItemPickaxe(18, 1).setIconIndex(97);
-        Item.AXE_STONE = new ItemAxe(19, 1).setIconIndex(113);
-        Item.SWORD_DIAMOND = new ItemSword(20, 3).setIconIndex(67);
-        Item.SHOVEL_DIAMOND = new ItemShovel(21, 3).setIconIndex(83);
-        Item.PICKAXE_DIAMOND = new ItemPickaxe(22, 3).setIconIndex(99);
-        Item.AXE_DIAMOND = new ItemAxe(23, 3).setIconIndex(115);
-        Item.STICK = new Item(24).setIconIndex(53).setFull3D();
-        Item.BOWL_EMPTY = new Item(25).setIconIndex(71);
-        Item.BOWL_SOUP = new ItemSoup(26, 10).setIconIndex(72);
-        Item.SWORD_GOLD = new ItemSword(27, 0).setIconIndex(68);
-        Item.SHOVEL_GOLD = new ItemShovel(28, 0).setIconIndex(84);
-        Item.PICKAXE_GOLD = new ItemPickaxe(29, 0).setIconIndex(100);
-        Item.AXE_GOLD = new ItemAxe(30, 0).setIconIndex(116);
-        Item.STRING = new Item(31).setIconIndex(8);
-        Item.FEATHER = new Item(32).setIconIndex(24);
-        Item.GUNPOWDER = new Item(33).setIconIndex(40);
-        Item.HOE_WOOD = new ItemHoe(34, 0).setIconIndex(128);
-        Item.HOE_STONE = new ItemHoe(35, 1).setIconIndex(129);
-        Item.HOE_IRON = new ItemHoe(36, 2).setIconIndex(130);
-        Item.HOE_DIAMOND = new ItemHoe(37, 3).setIconIndex(131);
-        Item.HOE_GOLD = new ItemHoe(38, 1).setIconIndex(132);
-        Item.SEEDS = new ItemSeeds(39, Block.CROPS.blockID).setIconIndex(9);
-        Item.WHEAT = new Item(40).setIconIndex(25);
-        Item.BREAD = new ItemFood(41, 5).setIconIndex(41);
-        Item.HELMET_LEATHER = new ItemArmor(42, 0, 0, 0).setIconIndex(0);
-        Item.PLATE_LEATHER = new ItemArmor(43, 0, 0, 1).setIconIndex(16);
-        Item.LEGS_LEATHER = new ItemArmor(44, 0, 0, 2).setIconIndex(32);
-        Item.BOOTS_LEATHER = new ItemArmor(45, 0, 0, 3).setIconIndex(48);
-        Item.HELMET_CHAIN = new ItemArmor(46, 1, 1, 0).setIconIndex(1);
-        Item.PLATE_CHAIN = new ItemArmor(47, 1, 1, 1).setIconIndex(17);
-        Item.LEGS_CHAIN = new ItemArmor(48, 1, 1, 2).setIconIndex(33);
-        Item.BOOTS_CHAIN = new ItemArmor(49, 1, 1, 3).setIconIndex(49);
-        Item.HELMET_IRON = new ItemArmor(50, 2, 2, 0).setIconIndex(2);
-        Item.PLATE_IRON = new ItemArmor(51, 2, 2, 1).setIconIndex(18);
-        Item.LEGS_IRON = new ItemArmor(52, 2, 2, 2).setIconIndex(34);
-        Item.BOOTS_IRON = new ItemArmor(53, 2, 2, 3).setIconIndex(50);
-        Item.HELMET_DIAMOND = new ItemArmor(54, 3, 3, 0).setIconIndex(3);
-        Item.PLATE_DIAMOND = new ItemArmor(55, 3, 3, 1).setIconIndex(19);
-        Item.LEGS_DIAMOND = new ItemArmor(56, 3, 3, 2).setIconIndex(35);
-        Item.BOOTS_DIAMOND = new ItemArmor(57, 3, 3, 3).setIconIndex(51);
-        Item.HELMET_GOLD = new ItemArmor(58, 1, 4, 0).setIconIndex(4);
-        Item.PLATE_GOLD = new ItemArmor(59, 1, 4, 1).setIconIndex(20);
-        Item.LEGS_GOLD = new ItemArmor(60, 1, 4, 2).setIconIndex(36);
-        Item.BOOTS_GOLD = new ItemArmor(61, 1, 4, 3).setIconIndex(52);
-        Item.FLINT = new Item(62).setIconIndex(6);
-        Item.PORK_RAW = new ItemFood(63, 3).setIconIndex(87);
-        Item.PORK_COOKED = new ItemFood(64, 8).setIconIndex(88);
-        Item.PAINTING = new ItemPainting(65).setIconIndex(26);
-        Item.APPLE_GOLD = new ItemFood(66, 42).setIconIndex(11);
-        Item.SIGN = new ItemSign(67).setIconIndex(42);
-        Item.DOOR_WOOD = new ItemDoor(68, Material.WOOD).setIconIndex(43);
-        Item.BUCKET_EMPTY = new ItemBucket(69, 0).setIconIndex(74);
-        Item.BUCKET_WATER = new ItemBucket(70, Block.WATER_MOVING.blockID).setIconIndex(75);
-        Item.BUCKET_LAVA = new ItemBucket(71, Block.LAVA_MOVING.blockID).setIconIndex(76);
-        Item.MINECART_EMPTY = new ItemMinecart(72, 0).setIconIndex(135);
-        Item.SADDLE = new ItemSaddle(73).setIconIndex(104);
-        Item.DOOR_IRON = new ItemDoor(74, Material.IRON).setIconIndex(44);
-        Item.REDSTONE = new ItemRedstone(75).setIconIndex(56);
-        Item.SNOWBALL = new ItemSnowball(76).setIconIndex(14);
-        Item.BOAT = new ItemBoat(77).setIconIndex(136);
-        Item.LEATHER = new Item(78).setIconIndex(103);
-        Item.BUCKET_MILK = new ItemBucket(79, -1).setIconIndex(77);
-        Item.BRICK = new Item(80).setIconIndex(22);
-        Item.CLAY = new Item(81).setIconIndex(57);
-        Item.REED = new ItemReed(82, Block.REED).setIconIndex(27);
-        Item.PAPER = new Item(83).setIconIndex(58);
-        Item.BOOK = new Item(84).setIconIndex(59);
-        Item.SLIMEBALL = new Item(85).setIconIndex(30);
-        Item.MINECART_CHEST = new ItemMinecart(86, 1).setIconIndex(151);
-        Item.MINECART_FURNACE = new ItemMinecart(87, 2).setIconIndex(167);
-        Item.EGG = new Item(88).setIconIndex(12);
-        Item.COMPASS = new Item(89).setIconIndex(54);
-        Item.RECORD_HIDDEN_DEN = new ItemRecord(2000, "hidden_den").setIconIndex(240);
-        Item.RECORD_LEMURIA = new ItemRecord(2001, "lemuria").setIconIndex(241);
-        Item.obsidianArmor0 = new ItemArmor(90, 4, 5, 0).setIconIndex(144);
-        Item.obsidianArmor1 = new ItemArmor(91, 4, 5, 1).setIconIndex(160);
-        Item.obsidianArmor2 = new ItemArmor(92, 4, 5, 2).setIconIndex(176);
-        Item.obsidianArmor3 = new ItemArmor(93, 4, 5, 3).setIconIndex(192);
-        Item.obsidianSword = new ItemSword(94, 5).setIconIndex(145);
-        Item.obsidianShovel = new ItemShovel(95, 5).setIconIndex(161);
-        Item.obsidianPick = new ObsidianPick(96, 5).setIconIndex(177);
-        Item.obsidianAxe = new ItemAxe(97, 5).setIconIndex(193);
-        Item.obsidianHoe = new ItemHoe(98, 5).setIconIndex(209);
-        Item.obsidianIngot = new Item(99).setIconIndex(208);
-        Item.blackDye = new Item(100).setIconIndex(146);
-        Item.greenDye = new Item(101).setIconIndex(147);
-        Item.blueDye = new Item(102).setIconIndex(148);
-        Item.pinkDye = new Item(103).setIconIndex(149);
-        Item.fryshroom = new ItemFood(104, 6).setIconIndex(162);
-        Item.edibleFire = new ItemFood(105, 2).setIconIndex(178);
-        Item.flameberge = new ItemSwordFlameberge(106).setIconIndex(69);
+
+    public final int shiftedIndex;
+    protected int maxStackSize;
+    protected int maxDamage;
+    protected int iconIndex;
+    protected boolean bFull3D;
+    protected String itemName = null;
+    protected int rarity = 0;
+    protected boolean isVanilla = true;
+
+    public String getName() {
+        if (itemName == null) return "item." + this.shiftedIndex + ".name";
+        return itemName;
     }
+
+    public Item setName(String itemName) {
+        this.itemName = itemName;
+        return this;
+    }
+
+    public int getRarity() {
+        return this.rarity;
+    }
+
+    public Item setRarity(int rarity) {
+        this.rarity = rarity;
+        return this;
+    }
+
+    public boolean getVanilla() {
+        return isVanilla;
+    }
+
+    public Item setVanilla(boolean isVanilla) {
+        this.isVanilla = isVanilla;
+        return this;
+    }
+
+    public int getID() {
+        return shiftedIndex;
+    }
+
+    public static void initAll() {
+        Item.SHOVEL_IRON = new ItemShovel(0, 2).setIconIndex(82).setName("Iron shovel");
+        Item.PICKAXE_IRON = new ItemPickaxe(1, 2).setIconIndex(98).setName("Iron pickaxe");
+        Item.AXE_IRON = new ItemAxe(2, 2).setIconIndex(114).setName("Iron axe");
+        Item.FLINT_AND_STEEL = new ItemFlintAndSteel(3).setIconIndex(5).setName("Flint and steel");
+        Item.APPLE = new ItemFood(4, 4).setIconIndex(10).setName("Apple");
+        Item.BOW = new ItemBow(5).setIconIndex(21).setName("Bow");
+        Item.ARROW = new Item(6).setIconIndex(37).setName("Arrow");
+        Item.COAL = new Item(7).setIconIndex(7).setName("Coal");
+        Item.DIAMOND = new Item(8).setIconIndex(55).setName("Diamond").setRarity(1);
+        Item.INGOT_IRON = new Item(9).setIconIndex(23).setName("Iron ingot");
+        Item.INGOT_GOLD = new Item(10).setIconIndex(39).setName("Golden ingot");
+        Item.SWORD_IRON = new ItemSword(11, 2).setIconIndex(66).setName("Iron sword");
+        Item.SWORD_WOOD = new ItemSword(12, 0).setIconIndex(64).setName("Wooden sword");
+        Item.SHOVEL_WOOD = new ItemShovel(13, 0).setIconIndex(80).setName("Wooden shovel");
+        Item.PICKAXE_WOOD = new ItemPickaxe(14, 0).setIconIndex(96).setName("Wooden pickaxe");
+        Item.AXE_WOOD = new ItemAxe(15, 0).setIconIndex(112).setName("Wooden axe");
+        Item.SWORD_STONE = new ItemSword(16, 1).setIconIndex(65).setName("Stone sword");
+        Item.SHOVEL_STONE = new ItemShovel(17, 1).setIconIndex(81).setName("Stone shovel");
+        Item.PICKAXE_STONE = new ItemPickaxe(18, 1).setIconIndex(97).setName("Stone pickaxe");
+        Item.AXE_STONE = new ItemAxe(19, 1).setIconIndex(113).setName("Stone axe");
+        Item.SWORD_DIAMOND = new ItemSword(20, 3).setIconIndex(67).setName("Diamond sword").setRarity(1);
+        Item.SHOVEL_DIAMOND = new ItemShovel(21, 3).setIconIndex(83).setName("Diamond shovel").setRarity(1);
+        Item.PICKAXE_DIAMOND = new ItemPickaxe(22, 3).setIconIndex(99).setName("Diamond pickaxe").setRarity(1);
+        Item.AXE_DIAMOND = new ItemAxe(23, 3).setIconIndex(115).setName("Diamond axe").setRarity(1);
+        Item.STICK = new Item(24).setIconIndex(53).setFull3D().setName("Stick");
+        Item.BOWL_EMPTY = new Item(25).setIconIndex(71).setName("Bowl");
+        Item.BOWL_SOUP = new ItemSoup(26, 10).setIconIndex(72).setName("Soup");
+        Item.SWORD_GOLD = new ItemSword(27, 0).setIconIndex(68).setName("Golden sword");
+        Item.SHOVEL_GOLD = new ItemShovel(28, 0).setIconIndex(84).setName("Golden shovel");
+        Item.PICKAXE_GOLD = new ItemPickaxe(29, 0).setIconIndex(100).setName("Golden pickaxe");
+        Item.AXE_GOLD = new ItemAxe(30, 0).setIconIndex(116).setName("Golden axe");
+        Item.STRING = new Item(31).setIconIndex(8).setName("String");
+        Item.FEATHER = new Item(32).setIconIndex(24).setName("Feather");
+        Item.GUNPOWDER = new Item(33).setIconIndex(40).setName("Gunpowder");
+        Item.HOE_WOOD = new ItemHoe(34, 0).setIconIndex(128).setName("Wooden hoe");
+        Item.HOE_STONE = new ItemHoe(35, 1).setIconIndex(129).setName("Stone hoe");
+        Item.HOE_IRON = new ItemHoe(36, 2).setIconIndex(130).setName("Iron hoe");
+        Item.HOE_DIAMOND = new ItemHoe(37, 3).setIconIndex(131).setName("Diamond hoe");
+        Item.HOE_GOLD = new ItemHoe(38, 1).setIconIndex(132).setName("Golden hoe");
+        Item.SEEDS = new ItemSeeds(39, Block.CROPS.blockID).setIconIndex(9).setName("Seeds");
+        Item.WHEAT = new Item(40).setIconIndex(25).setName("Wheat");
+        Item.BREAD = new ItemFood(41, 5).setIconIndex(41).setName("Bread");
+        Item.HELMET_LEATHER = new ItemArmor(42, 0, 0, 0).setIconIndex(0).setName("Leather hat");
+        Item.PLATE_LEATHER = new ItemArmor(43, 0, 0, 1).setIconIndex(16).setName("Leather tunic");
+        Item.LEGS_LEATHER = new ItemArmor(44, 0, 0, 2).setIconIndex(32).setName("Leather pants");
+        Item.BOOTS_LEATHER = new ItemArmor(45, 0, 0, 3).setIconIndex(48).setName("Leather boots");
+        Item.HELMET_CHAIN = new ItemArmor(46, 1, 1, 0).setIconIndex(1).setName("Chainmail helmet");
+        Item.PLATE_CHAIN = new ItemArmor(47, 1, 1, 1).setIconIndex(17).setName("Chainmail chestplate");
+        Item.LEGS_CHAIN = new ItemArmor(48, 1, 1, 2).setIconIndex(33).setName("Chainmail leggings");
+        Item.BOOTS_CHAIN = new ItemArmor(49, 1, 1, 3).setIconIndex(49).setName("Chainmail boots");
+        Item.HELMET_IRON = new ItemArmor(50, 2, 2, 0).setIconIndex(2).setName("Iron helmet");
+        Item.PLATE_IRON = new ItemArmor(51, 2, 2, 1).setIconIndex(18).setName("Iron chestplate");
+        Item.LEGS_IRON = new ItemArmor(52, 2, 2, 2).setIconIndex(34).setName("Iron leggings");
+        Item.BOOTS_IRON = new ItemArmor(53, 2, 2, 3).setIconIndex(50).setName("Iron boots");
+        Item.HELMET_DIAMOND = new ItemArmor(54, 3, 3, 0).setIconIndex(3).setName("Diamond helmet").setRarity(1);
+        Item.PLATE_DIAMOND = new ItemArmor(55, 3, 3, 1).setIconIndex(19).setName("Diamond leggings").setRarity(1);
+        Item.LEGS_DIAMOND = new ItemArmor(56, 3, 3, 2).setIconIndex(35).setName("Diamond leggings").setRarity(1);
+        Item.BOOTS_DIAMOND = new ItemArmor(57, 3, 3, 3).setIconIndex(51).setName("Diamond boots").setRarity(1);
+        Item.HELMET_GOLD = new ItemArmor(58, 1, 4, 0).setIconIndex(4).setName("Golden helmet");
+        Item.PLATE_GOLD = new ItemArmor(59, 1, 4, 1).setIconIndex(20).setName("Golden chestplate");
+        Item.LEGS_GOLD = new ItemArmor(60, 1, 4, 2).setIconIndex(36).setName("Golden leggings");
+        Item.BOOTS_GOLD = new ItemArmor(61, 1, 4, 3).setIconIndex(52).setName("Golden boots");
+        Item.FLINT = new Item(62).setIconIndex(6).setName("Flint");
+        Item.PORK_RAW = new ItemFood(63, 3).setIconIndex(87).setName("Raw porkchop");
+        Item.PORK_COOKED = new ItemFood(64, 8).setIconIndex(88).setName("Steak");
+        Item.PAINTING = new ItemPainting(65).setIconIndex(26).setName("Crash engine TM");
+        Item.APPLE_GOLD = new ItemFood(66, 42).setIconIndex(11).setName("Golden apple").setRarity(2);
+        Item.SIGN = new ItemSign(67).setIconIndex(42).setName("Sign");
+        Item.DOOR_WOOD = new ItemDoor(68, Material.WOOD).setIconIndex(43).setName("Wooden door");
+        Item.BUCKET_EMPTY = new ItemBucket(69, 0).setIconIndex(74).setName("Bucket of air");
+        Item.BUCKET_WATER = new ItemBucket(70, Block.WATER_MOVING.blockID).setIconIndex(75).setName("Bucket of water");
+        Item.BUCKET_LAVA = new ItemBucket(71, Block.LAVA_MOVING.blockID).setIconIndex(76).setName("Bucket of lava");
+        Item.MINECART_EMPTY = new ItemMinecart(72, 0).setIconIndex(135).setName("Don't use, doesn't work").setRarity(-1);
+        Item.SADDLE = new ItemSaddle(73).setIconIndex(104).setName("Saddle");
+        Item.DOOR_IRON = new ItemDoor(74, Material.IRON).setIconIndex(44).setName("Hub door");
+        Item.REDSTONE = new ItemRedstone(75).setIconIndex(56).setName("Greenstone");
+        Item.SNOWBALL = new ItemSnowball(76).setIconIndex(14).setName("Snowball");
+        Item.BOAT = new ItemBoat(77).setIconIndex(136).setName("Boat");
+        Item.LEATHER = new Item(78).setIconIndex(103).setName("Leather");
+        Item.BUCKET_MILK = new ItemBucket(79, -1).setIconIndex(77).setName("You dad's purchase");
+        Item.BRICK = new Item(80).setIconIndex(22).setName("Brick");
+        Item.CLAY = new Item(81).setIconIndex(57).setName("Dream");
+        Item.REED = new ItemReed(82, Block.REED).setIconIndex(27).setName("Sugar reeds");
+        Item.PAPER = new Item(83).setIconIndex(58).setName("Paper");
+        Item.BOOK = new Item(84).setIconIndex(59).setName("Book");
+        Item.SLIMEBALL = new Item(85).setIconIndex(30).setName("Slimeball");
+        Item.MINECART_CHEST = new ItemMinecart(86, 1).setIconIndex(151).setName("Chest minecart");
+        Item.MINECART_FURNACE = new ItemMinecart(87, 2).setIconIndex(167).setName("Coal burner TM");
+        Item.EGG = new Item(88).setIconIndex(12).setName("Egg").setRarity(2);
+        Item.COMPASS = new Item(89).setIconIndex(54).setName("Compass");
+        Item.RECORD_HIDDEN_DEN = new ItemRecord(2000, "hidden_den").setIconIndex(240).setName("[5] - Hidden Den").setRarity(2);
+        Item.RECORD_LEMURIA = new ItemRecord(2001, "lemuria").setIconIndex(241).setName("[5] - Lemuria").setRarity(2);
+        Item.RECORD_43343138 = new ItemRecord(2002, "43343138").setIconIndex(242).setName("[5] - 43343138").setRarity(2).setVanilla(false);
+        Item.RECORD_ESTONIAN_RELIQUARY = new ItemRecord(2003, "estonian_reliquary").setIconIndex(243).setName("[5] - Estonian Reliquary").setRarity(2).setVanilla(false);
+        Item.obsidianArmor0 = new ItemArmor(90, 4, 5, 0).setIconIndex(144).setName("Obsidian helmet").setRarity(1);
+        Item.obsidianArmor1 = new ItemArmor(91, 4, 5, 1).setIconIndex(160).setName("Obsidian chestplate").setRarity(1);
+        Item.obsidianArmor2 = new ItemArmor(92, 4, 5, 2).setIconIndex(176).setName("Obsidian leggings").setRarity(1);
+        Item.obsidianArmor3 = new ItemArmor(93, 4, 5, 3).setIconIndex(192).setName("Obsidian boots").setRarity(1);
+        Item.obsidianSword = new ItemSword(94, 5).setIconIndex(145).setName("Obsidian sword").setRarity(1);
+        Item.obsidianShovel = new ItemShovel(95, 5).setIconIndex(161).setName("Obsidian shovel").setRarity(1);
+        Item.obsidianPick = new ObsidianPick(96, 5).setIconIndex(177).setName("Obsidian pickaxe").setRarity(1);
+        Item.obsidianAxe = new ItemAxe(97, 5).setIconIndex(193).setName("Obsidian axe").setRarity(1);
+        Item.obsidianHoe = new ItemHoe(98, 5).setIconIndex(209).setName("Obsidian hoe").setRarity(1);
+        Item.obsidianIngot = new Item(99).setIconIndex(208).setName("Obsidian ingot").setRarity(1);
+        Item.blackDye = new Item(100).setIconIndex(146).setName("Black dye");
+        Item.greenDye = new Item(101).setIconIndex(147).setName("Green dye");
+        Item.blueDye = new Item(102).setIconIndex(148).setName("Blue dye");
+        Item.pinkDye = new Item(103).setIconIndex(149).setName("Pink dye");
+        Item.fryshroom = new ItemFood(104, 6).setIconIndex(162).setName("Fryshroom");
+        Item.edibleFire = new ItemFood(105, 2).setIconIndex(178).setName("Fire but ediable");
+        Item.flameberge = new ItemSwordFlameberge(106).setIconIndex(69).setName("Flameberge").setRarity(2);
+    }
     
     protected Item(final int id) {
@@ -241,5 +280,5 @@
         this.shiftedIndex = 256 + id;
         if (Item.ITEMS_LIST[256 + id] != null) {
-            System.out.println(new StringBuilder().append("CONFLICT @ ").append(id).toString());
+            System.out.println("CONFLICT @ " + id);
         }
         Item.ITEMS_LIST[256 + id] = this;
@@ -285,5 +324,5 @@
     }
     
-    public boolean canHarvestBlock(final Block block) {
+    public boolean canHarvestBlock(final Block block, final World world) {
         return false;
     }
