--- MinecraftAppletImpl.java
+++ MinecraftAppletImpl.java
@@ -1,4 +1,5 @@
 package net.minecraft.src;
 
+import java.applet.Applet;
 import java.awt.LayoutManager;
 import java.awt.BorderLayout;
@@ -9,16 +10,16 @@
 
 public class MinecraftAppletImpl extends Minecraft {
-    final MinecraftApplet mainFrame;
-    
-    public MinecraftAppletImpl(final MinecraftApplet frame, final Component component, final Canvas canvas, final MinecraftApplet mcApplet, final int width, final int height, final boolean fullscreen) {
+    final Applet mainFrame;
+
+    public MinecraftAppletImpl(final Applet frame, final Component component, final Canvas canvas, final Applet mcApplet, final int width, final int height, final boolean fullscreen) {
         super(component, canvas, mcApplet, width, height, fullscreen);
         this.mainFrame = frame;
     }
-    
+
     @Override
     public void displayUnexpectedThrowable(final UnexpectedThrowable unexpectedThrowable) {
         this.mainFrame.removeAll();
-        this.mainFrame.setLayout((LayoutManager)new BorderLayout());
-        this.mainFrame.add((Component)new PanelCrashReport(unexpectedThrowable), "Center");
+        this.mainFrame.setLayout(new BorderLayout());
+        this.mainFrame.add(new PanelCrashReport(unexpectedThrowable), "Center");
         this.mainFrame.validate();
     }
