--- MovementInputFromOptions.java
+++ MovementInputFromOptions.java
@@ -1,8 +1,10 @@
 package net.minecraft.src;
 
+import net.buj.rml.Game;
+
 public class MovementInputFromOptions extends MovementInput {
     private boolean[] movementKeyStates;
     private GameSettings gameSettings;
-    
+
     public MovementInputFromOptions(final GameSettings gameSettings) {
         InputHandler.CaptureGamepad();
@@ -10,21 +12,21 @@
         this.gameSettings = gameSettings;
     }
-    
+
     @Override
     public void checkKeyForMovementInput(final int key, final boolean state) {
         int n = -1;
-        if (key == this.gameSettings.keyBindForward.keyCode) {
+        if (key == Game.OPTIONS.keyCodeFor("forward")) {
             n = 0;
         }
-        if (key == this.gameSettings.keyBindBack.keyCode) {
+        if (key == Game.OPTIONS.keyCodeFor("back")) {
             n = 1;
         }
-        if (key == this.gameSettings.keyBindLeft.keyCode) {
+        if (key == Game.OPTIONS.keyCodeFor("left")) {
             n = 2;
         }
-        if (key == this.gameSettings.keyBindRight.keyCode) {
+        if (key == Game.OPTIONS.keyCodeFor("right")) {
             n = 3;
         }
-        if (key == this.gameSettings.keyBindJump.keyCode) {
+        if (key == Game.OPTIONS.keyCodeFor("jump")) {
             n = 4;
         }
@@ -33,5 +35,5 @@
         }
     }
-    
+
     @Override
     public void resetKeyState() {
@@ -40,5 +42,5 @@
         }
     }
-    
+
     @Override
     public void updatePlayerMoveState(final EntityPlayer entityPlayer) {
