--- NetworkManager.java
+++ NetworkManager.java
@@ -1,4 +1,7 @@
 package net.minecraft.src;
 
+import net.minecraft.client.Minecraft;
+import org.lwjgl.Sys;
+
 import java.io.IOException;
 import java.util.Collections;
@@ -13,12 +16,12 @@
     public static int numReadThreads;
     public static int numWriteThreads;
-    private Object sendQueueLock;
-    private Socket networkSocket;
-    private DataInputStream socketInputStream;
-    private DataOutputStream socketOutputStream;
-    boolean isRunning;
-    private List<Packet> readPackets;
-    private List<Packet> dataPackets;
-    private List<Packet> chunkDataPackets;
+    private final Object sendQueueLock;
+    private final Socket networkSocket;
+    private final DataInputStream socketInputStream;
+    private final DataOutputStream socketOutputStream;
+    public boolean isRunning;
+    private final List<Packet> readPackets;
+    private final List<Packet> dataPackets;
+    private final List<Packet> chunkDataPackets;
     private NetHandler netHandler;
     boolean isServerTerminating;
@@ -31,16 +34,32 @@
     private int chunkDataSendCounter;
     private boolean isSwitchingHandler;
+    private Minecraft mc;
+    public boolean graceful = false;
+
+    public final ThreadNetworkKeepAlive keepAlive;
     
     static {
         threadSyncObject = new Object();
     }
+
+    public boolean getIsRunning() {
+        synchronized (threadSyncObject) {
+            return isRunning;
+        }
+    }
+
+    public NetHandler getNetHandler() {
+        synchronized (threadSyncObject) {
+            return netHandler;
+        }
+    }
     
-    public NetworkManager(final Socket socket, final String name, final NetHandler netHandler) throws IOException {
+    public NetworkManager(final Minecraft mc, final Socket socket, final String name, final NetHandler netHandler) throws IOException {
         this.isSwitchingHandler = false;
         this.sendQueueLock = new Object();
         this.isRunning = true;
-        this.readPackets = (List<Packet>)Collections.synchronizedList((List)new LinkedList());
-        this.dataPackets = (List<Packet>)Collections.synchronizedList((List)new LinkedList());
-        this.chunkDataPackets = (List<Packet>)Collections.synchronizedList((List)new LinkedList());
+        this.readPackets = Collections.synchronizedList(new LinkedList<>());
+        this.dataPackets = Collections.synchronizedList(new LinkedList<>());
+        this.chunkDataPackets = Collections.synchronizedList(new LinkedList<>());
         this.isServerTerminating = false;
         this.isTerminating = false;
@@ -51,11 +70,15 @@
         this.networkSocket = socket;
         this.netHandler = netHandler;
+        this.mc = mc;
         socket.setTrafficClass(24);
-        this.socketInputStream = new DataInputStream(socket.getInputStream());
-        this.socketOutputStream = new DataOutputStream(socket.getOutputStream());
-        this.readThread = new NetworkReaderThread(this, String.valueOf(name) + " read thread");
-        this.writeThread = new NetworkWriterThread(this, String.valueOf(name) + " write thread");
+        this.socketInputStream = new CountDataInputStream(socket.getInputStream());
+        this.socketOutputStream = new CountDataOutputStream(socket.getOutputStream());
+        this.keepAlive = new ThreadNetworkKeepAlive(this);
+        this.readThread = new NetworkReaderThread(keepAlive, this, name + " read thread");
+        this.writeThread = new NetworkWriterThread(this, name + " write thread");
         this.readThread.start();
         this.writeThread.start();
+        this.keepAlive.start();
+        ThreadBandwidthCounter.spawn();
     }
     
@@ -73,5 +96,4 @@
             }
         }
-        // monitorexit(this.sendQueueLock)
     }
     
@@ -92,8 +114,7 @@
                 final Packet packet;
                 synchronized (this.sendQueueLock) {
-                    packet = (Packet)this.dataPackets.remove(0);
+                    packet = this.dataPackets.remove(0);
                     this.sendQueueByteLength -= packet.getPacketSize() + 1;
                 }
-                // monitorexit(this.sendQueueLock)
                 Packet.writePacket(packet, this.socketOutputStream);
             }
@@ -102,8 +123,7 @@
                 final Packet packet;
                 synchronized (this.sendQueueLock) {
-                    packet = (Packet)this.chunkDataPackets.remove(0);
+                    packet = this.chunkDataPackets.remove(0);
                     this.sendQueueByteLength -= packet.getPacketSize() + 1;
                 }
-                // monitorexit(this.sendQueueLock)
                 Packet.writePacket(packet, this.socketOutputStream);
                 this.chunkDataSendCounter = 50;
@@ -113,5 +133,5 @@
             }
         }
-        catch (InterruptedException ex) {}
+        catch (InterruptedException ignored) {}
         catch (Exception exception) {
             if (!this.isTerminating) {
@@ -129,4 +149,7 @@
             else {
                 this.networkShutdown("End of stream");
+                if (!graceful) {
+                    this.mc.displayGuiScreen(new GuiConnectFailed("Connection closed", "End of stream"));
+                }
             }
         }
@@ -140,8 +163,9 @@
     private void onNetworkError(final Exception exception) {
         exception.printStackTrace();
-        this.networkShutdown(new StringBuilder("Internal exception: ").append(exception.toString()).toString());
+        this.networkShutdown("Internal exception: " + exception.toString());
     }
     
     public void networkShutdown(final String reason) {
+        System.out.println("Network closed: " + reason);
         if (!this.isRunning) {
             return;
@@ -154,13 +178,13 @@
             this.socketInputStream.close();
         }
-        catch (Throwable t) {}
+        catch (Throwable ignored) {}
         try {
             this.socketOutputStream.close();
         }
-        catch (Throwable t2) {}
+        catch (Throwable ignored) {}
         try {
             this.networkSocket.close();
         }
-        catch (Throwable t3) {}
+        catch (Throwable ignored) {}
     }
     
@@ -171,8 +195,12 @@
         if (this.sendQueueByteLength > 1048576) {
             this.networkShutdown("Send buffer overflow");
+            this.mc.changeWorld1(null);
+            this.mc.displayGuiScreen(new GuiConnectFailed("Connection closed", "Send buffer overflow"));
         }
         if (this.readPackets.isEmpty()) {
             if (this.timeSinceLastRead++ == 1200) {
                 this.networkShutdown("Timed out");
+                this.mc.changeWorld1(null);
+                this.mc.displayGuiScreen(new GuiConnectFailed("Connection closed", "Timed out"));
             }
         }
@@ -182,5 +210,5 @@
         int n = 100;
         while (!this.readPackets.isEmpty() && n-- >= 0 && !this.isSwitchingHandler) {
-            ((Packet)this.readPackets.remove(0)).processPacket(this.netHandler);
+            this.readPackets.remove(0).processPacket(this.netHandler);
         }
         if (this.isTerminating && this.readPackets.isEmpty()) {
