--- RecipesTools.java
+++ RecipesTools.java
@@ -7,5 +7,5 @@
     public RecipesTools() {
         this.recipePatterns = new String[][] { { "XXX", " # ", " # " }, { "X", "#", "#" }, { "XX", "X#", " #" }, { "XX", " #", " #" } };
-        this.recipeItems = new Object[][] { { Block.PLANKS, Block.COBBLESTONE, Item.INGOT_IRON, Item.DIAMOND, Item.INGOT_GOLD }, { Item.PICKAXE_WOOD, Item.PICKAXE_STONE, Item.PICKAXE_IRON, Item.PICKAXE_DIAMOND, Item.PICKAXE_GOLD }, { Item.SHOVEL_WOOD, Item.SHOVEL_STONE, Item.SHOVEL, Item.SHOVEL_DIAMOND, Item.SHOVEL_GOLD }, { Item.AXE_WOOD, Item.AXE_STONE, Item.AXE_IRON, Item.AXE_DIAMOND, Item.AXE_GOLD }, { Item.HOE_WOOD, Item.HOE_STONE, Item.HOE_IRON, Item.HOE_DIAMOND, Item.HOE_GOLD } };
+        this.recipeItems = new Object[][] { { Block.PLANKS, Block.COBBLESTONE, Item.INGOT_IRON, Item.DIAMOND, Item.INGOT_GOLD }, { Item.PICKAXE_WOOD, Item.PICKAXE_STONE, Item.PICKAXE_IRON, Item.PICKAXE_DIAMOND, Item.PICKAXE_GOLD }, { Item.SHOVEL_WOOD, Item.SHOVEL_STONE, Item.SHOVEL_IRON, Item.SHOVEL_DIAMOND, Item.SHOVEL_GOLD }, { Item.AXE_WOOD, Item.AXE_STONE, Item.AXE_IRON, Item.AXE_DIAMOND, Item.AXE_GOLD }, { Item.HOE_WOOD, Item.HOE_STONE, Item.HOE_IRON, Item.HOE_DIAMOND, Item.HOE_GOLD } };
     }
     
