--- RenderManager.java
+++ RenderManager.java
@@ -57,8 +57,10 @@
     }
     
-    public Render getEntityClassRendderObject(final Class<?> entityClass) {
-        Render entityClassRendderObject = (Render)this.entityRenderMap.get(entityClass);
+    public Render getEntityClassRendderObject(final Class<? extends Entity> entityClass) {
+        Render entityClassRendderObject = this.entityRenderMap.get(entityClass);
         if (entityClassRendderObject == null && entityClass != Entity.class) {
-            entityClassRendderObject = this.getEntityClassRendderObject(entityClass.getSuperclass());
+            entityClassRendderObject = this.getEntityClassRendderObject(
+            		(Class<? extends Entity>) entityClass.getSuperclass()
+            );
             this.entityRenderMap.put(entityClass, entityClassRendderObject);
         }
