--- TextureFlamesFX.java
+++ TextureFlamesFX.java
@@ -4,5 +4,5 @@
     protected float[] currentFireFrame;
     protected float[] lastFireFrame;
-    
+
     public TextureFlamesFX(final int index) {
         super(Block.FIRE.blockIndexInTexture + index * 16);
@@ -10,5 +10,5 @@
         this.lastFireFrame = new float[320];
     }
-    
+
     @Override
     public void onTick() {
@@ -61,7 +61,13 @@
                 n3 = n8;
             }
-            this.imageData[j * 4 + 0] = (byte)k;
+            if (RenderEngine.reverseColors) {
+                this.imageData[j * 4 + 0] = (byte)n3;
+                this.imageData[j * 4 + 2] = (byte)k;
+            }
+            else {
+                this.imageData[j * 4 + 0] = (byte)k;
+                this.imageData[j * 4 + 2] = (byte)n3;
+            }
             this.imageData[j * 4 + 1] = (byte)l;
-            this.imageData[j * 4 + 2] = (byte)n3;
             this.imageData[j * 4 + 3] = (byte)n4;
         }
