--- ThreadDownloadImage.java
+++ ThreadDownloadImage.java
@@ -15,5 +15,5 @@
     private String buffer;
     private IImageBuffer imageData;
-    
+
     ThreadDownloadImage(final ThreadDownloadImageData downloadImageData, final String location, final IImageBuffer image) {
         this.location = downloadImageData;
@@ -21,15 +21,32 @@
         this.imageData = image;
     }
-    
+
+    public boolean checkUsername(String username) {
+    	if (username.length() < 2 || username.length() > 32) return false;
+    	String fine = "qwertyuiopasdfghjklzxcvbnnmQWERTYUIOPASDFGHJKLZXCVBNM1234567890-_";
+    	for (int i = 0; i < username.length(); i++) {
+    		boolean good = false;
+    		for (int o = 0; o < fine.length(); o++) {
+    			if (username.charAt(i) == fine.charAt(o)) {
+    				good = true;
+    				break;
+    			}
+    		}
+    		if (!good) return false;
+    	}
+    	return true;
+    }
+
     public void run() {
         if (this.buffer.startsWith("skin:///")) {
             HttpURLConnection skinURLConnection = null;
             final String skinUsername = this.buffer.substring(8);
+            if (!checkUsername(skinUsername)) return;
             try {
-                final URL uuidURL = new URL(new StringBuilder("https://api.mojang.com/users/profiles/minecraft/").append(skinUsername).toString());
+                final URL uuidURL = new URL("https://api.mojang.com/users/profiles/minecraft/" + skinUsername);
                 final JSONTokener uuidTokener = new JSONTokener(uuidURL.openStream());
                 final JSONObject uuidRoot = new JSONObject(uuidTokener);
                 final String uuid = uuidRoot.getString("id");
-                final URL profileURL = new URL(new StringBuilder("https://sessionserver.mojang.com/session/minecraft/profile/").append(uuid).toString());
+                final URL profileURL = new URL("https://sessionserver.mojang.com/session/minecraft/profile/" + uuid);
                 final JSONTokener profileTokener = new JSONTokener(profileURL.openStream());
                 final JSONObject profileRoot = new JSONObject(profileTokener);
@@ -42,6 +59,10 @@
                 skinURLConnection.setDoOutput(false);
                 skinURLConnection.connect();
+
+                System.out.println("Skin URI: " + skinURL);
+
                 if (this.buffer == null) {
                     this.location.image = ImageIO.read(skinURLConnection.getInputStream());
+                    System.out.printf("X: %s, Y: %s\n", this.location.image.getWidth(), this.location.image.getHeight());
                 }
                 else {
