--- TileEntityRenderer.java
+++ TileEntityRenderer.java
@@ -7,5 +7,5 @@
 
 public class TileEntityRenderer {
-    private Map<Class<? extends TileEntity>, TileEntitySpecialRenderer> specialRendererMap;
+    private final Map<Class<? extends TileEntity>, TileEntitySpecialRenderer> specialRendererMap;
     public static TileEntityRenderer instance;
     private FontRenderer fontRenderer;
@@ -21,35 +21,34 @@
     public double playerY;
     public double playerZ;
-    
+
     static {
         TileEntityRenderer.instance = new TileEntityRenderer();
     }
-    
+
     private TileEntityRenderer() {
         (this.specialRendererMap = (Map<Class<? extends TileEntity>, TileEntitySpecialRenderer>)new HashMap()).put(TileEntitySign.class, new TileEntitySignRenderer());
         this.specialRendererMap.put(TileEntityMobSpawner.class, new TileEntityMobSpawnerRenderer());
-        final Iterator<TileEntitySpecialRenderer> iterator = (Iterator<TileEntitySpecialRenderer>)this.specialRendererMap.values().iterator();
-        while (iterator.hasNext()) {
-            ((TileEntitySpecialRenderer)iterator.next()).setTileEntityRenderer(this);
+        for (TileEntitySpecialRenderer tileEntitySpecialRenderer : this.specialRendererMap.values()) {
+            tileEntitySpecialRenderer.setTileEntityRenderer(this);
         }
     }
-    
+
     public TileEntitySpecialRenderer getSpecialRendererForClass(final Class<?> tileClass) {
-        TileEntitySpecialRenderer specialRendererForClass = (TileEntitySpecialRenderer)this.specialRendererMap.get(tileClass);
+        TileEntitySpecialRenderer specialRendererForClass = this.specialRendererMap.get(tileClass);
         if (specialRendererForClass == null && tileClass != TileEntity.class) {
             specialRendererForClass = this.getSpecialRendererForClass(tileClass.getSuperclass());
-            this.specialRendererMap.put(tileClass, specialRendererForClass);
+            this.specialRendererMap.put((Class<? extends TileEntity>) tileClass, specialRendererForClass);
         }
         return specialRendererForClass;
     }
-    
+
     public boolean hasSpecialRenderer(final TileEntity tileEntity) {
         return this.getSpecialRendererForEntity(tileEntity) != null;
     }
-    
+
     public TileEntitySpecialRenderer getSpecialRendererForEntity(final TileEntity tileEntity) {
         return this.getSpecialRendererForClass(tileEntity.getClass());
     }
-    
+
     public void cacheActiveRenderInfo(final World world, final RenderEngine renderEngine, final FontRenderer fontRenderer, final EntityPlayer entityPlayer, final float renderPartialTick) {
         this.worldObj = world;
@@ -63,13 +62,13 @@
         this.playerZ = entityPlayer.lastTickPosZ + (entityPlayer.posZ - entityPlayer.lastTickPosZ) * renderPartialTick;
     }
-    
+
     public void renderTileEntity(final TileEntity tileEntity, final float renderPartialTick) {
         if (tileEntity.getDistanceFrom(this.playerX, this.playerY, this.playerZ) < 4096.0) {
             final float brightness = this.worldObj.getBrightness(tileEntity.xCoord, tileEntity.yCoord, tileEntity.zCoord);
-            GL11.glColor3f(brightness, brightness, brightness);
+            RenderEngine.glColor3f(brightness, brightness, brightness);
             this.renderTileEntityAt(tileEntity, tileEntity.xCoord - TileEntityRenderer.staticPlayerX, tileEntity.yCoord - TileEntityRenderer.staticPlayerY, tileEntity.zCoord - TileEntityRenderer.staticPlayerZ, renderPartialTick);
         }
     }
-    
+
     public void renderTileEntityAt(final TileEntity tileEntity, final double x, final double y, final double z, final float renderPartialTick) {
         final TileEntitySpecialRenderer specialRendererForEntity = this.getSpecialRendererForEntity(tileEntity);
@@ -78,5 +77,5 @@
         }
     }
-    
+
     public FontRenderer getFontRenderer() {
         return this.fontRenderer;
