--- BlockIce.java
+++ BlockIce.java
@@ -19,5 +19,5 @@
         final Material material5 = world.getBlockMaterial(x, y - 1, z);
         if (material5.getIsSolid() || material5.getIsLiquid()) {
-            world.setBlockWithNotify(x, y, z, Block.waterMoving.blockID);
+            world.setBlockWithNotify(x, y, z, Block.WATER_MOVING.blockID);
         }
     }
@@ -32,5 +32,5 @@
         if (world.getSavedLightValue(EnumSkyBlock.Block, x, y, z) > 11 - Block.lightOpacity[this.blockID]) {
             this.dropBlockAsItem(world, x, y, z, world.getBlockMetadata(x, y, z));
-            world.setBlockWithNotify(x, y, z, Block.waterStill.blockID);
+            world.setBlockWithNotify(x, y, z, Block.WATER_STILL.blockID);
         }
     }
