--- EntityTrackerEntry.java
+++ EntityTrackerEntry.java
@@ -21,5 +21,5 @@
     private boolean firstUpdateDone;
     public boolean playerEntitiesUpdated;
-    public Set trackedPlayers;
+    public Set<EntityPlayerMP> trackedPlayers;
     
     public EntityTrackerEntry(final Entity entity1, final int i2, final int i3) {
@@ -67,5 +67,5 @@
             final int i8 = i3 - this.encodedPosY;
             final int i9 = i4 - this.encodedPosZ;
-            Object object12 = null;
+            Packet object12 = null;
             if (i7 >= -128 && i7 < 128 && i8 >= -128 && i8 < 128 && i9 >= -128 && i9 < 128) {
                 if (z7 && z8) {
@@ -78,7 +78,4 @@
                     object12 = new Packet32EntityLook(this.trackedEntity.entityID, (byte)i5, (byte)i6);
                 }
-                else {
-                    object12 = new Packet30Entity(this.trackedEntity.entityID);
-                }
             }
             else {
@@ -86,5 +83,5 @@
             }
             if (object12 != null) {
-                this.sendPacketToTrackedPlayers((Packet)object12);
+                this.sendPacketToTrackedPlayers(object12);
             }
             this.encodedPosX = i2;
