--- ItemFlintAndSteel.java
+++ ItemFlintAndSteel.java
@@ -31,5 +31,5 @@
         if (i8 == 0) {
             world.playSoundEffect(x + 0.5, y + 0.5, z + 0.5, "fire.ignite", 1.0f, ItemFlintAndSteel.rand.nextFloat() * 0.4f + 0.8f);
-            world.setBlockWithNotify(x, y, z, Block.fire.blockID);
+            world.setBlockWithNotify(x, y, z, Block.FIRE.blockID);
         }
         stack.damageItem(1);
