--- Item.java
+++ Item.java
@@ -4,6 +4,6 @@
 
 public class Item {
-    protected static Random rand;
-    public static Item[] itemsList;
+    protected static final Random rand = new Random();
+    public static Item[] itemsList = new Item[32000];
     public static Item shovel;
     public static Item pickaxeSteel;
@@ -98,4 +98,6 @@
     public static Item record13;
     public static Item recordCat;
+    public static Item record43343138;
+    public static Item recordEstonianReliquary;
     public final int shiftedIndex;
     protected int maxStackSize;
@@ -120,8 +122,7 @@
     public static Item edibleFire;
     public static Item flameberge;
-    
-    static {
-        Item.rand = new Random();
-        Item.itemsList = new Item[32000];
+
+
+    public static void initAll() {
         Item.shovel = new ItemSpade(0, 2).setIconIndex(82);
         Item.pickaxeSteel = new ItemPickaxe(1, 2).setIconIndex(98);
@@ -163,5 +164,5 @@
         Item.hoeDiamond = new ItemHoe(37, 3).setIconIndex(131);
         Item.hoeGold = new ItemHoe(38, 1).setIconIndex(132);
-        Item.seeds = new ItemSeeds(39, Block.crops.blockID).setIconIndex(9);
+        Item.seeds = new ItemSeeds(39, Block.CROPS.blockID).setIconIndex(9);
         Item.wheat = new Item(40).setIconIndex(25);
         Item.bread = new ItemFood(41, 5).setIconIndex(41);
@@ -194,6 +195,6 @@
         Item.doorWood = new ItemDoor(68, Material.wood).setIconIndex(43);
         Item.bucketEmpty = new ItemBucket(69, 0).setIconIndex(74);
-        Item.bucketWater = new ItemBucket(70, Block.waterMoving.blockID).setIconIndex(75);
-        Item.bucketLava = new ItemBucket(71, Block.lavaMoving.blockID).setIconIndex(76);
+        Item.bucketWater = new ItemBucket(70, Block.WATER_MOVING.blockID).setIconIndex(75);
+        Item.bucketLava = new ItemBucket(71, Block.LAVA_MOVING.blockID).setIconIndex(76);
         Item.minecartEmpty = new ItemMinecart(72, 0).setIconIndex(135);
         Item.saddle = new ItemSaddle(73).setIconIndex(104);
@@ -206,5 +207,5 @@
         Item.brick = new Item(80).setIconIndex(22);
         Item.clay = new Item(81).setIconIndex(57);
-        Item.reed = new ItemReed(82, Block.reed).setIconIndex(27);
+        Item.reed = new ItemReed(82, Block.REED).setIconIndex(27);
         Item.paper = new Item(83).setIconIndex(58);
         Item.book = new Item(84).setIconIndex(59);
@@ -216,4 +217,6 @@
         Item.record13 = new ItemRecord(2000, "hidden_den").setIconIndex(240);
         Item.recordCat = new ItemRecord(2001, "lemuria").setIconIndex(241);
+        Item.record43343138 = new ItemRecord(2002, "43343138").setIconIndex(242);
+        Item.recordEstonianReliquary = new ItemRecord(2003, "estonian_reliquary").setIconIndex(243);
         Item.obsidianArmor0 = new ItemArmor(90, 4, 5, 0).setIconIndex(144);
         Item.obsidianArmor1 = new ItemArmor(91, 4, 5, 1).setIconIndex(160);
