--- WorldGenLiquids.java
+++ WorldGenLiquids.java
@@ -12,24 +12,24 @@
     @Override
     public boolean generate(final World world, final Random rand, final int x, final int y, final int z) {
-        if (world.getBlockId(x, y + 1, z) != Block.stone.blockID) {
+        if (world.getBlockId(x, y + 1, z) != Block.STONE.blockID) {
             return false;
         }
-        if (world.getBlockId(x, y - 1, z) != Block.stone.blockID) {
+        if (world.getBlockId(x, y - 1, z) != Block.STONE.blockID) {
             return false;
         }
-        if (world.getBlockId(x, y, z) != 0 && world.getBlockId(x, y, z) != Block.stone.blockID) {
+        if (world.getBlockId(x, y, z) != 0 && world.getBlockId(x, y, z) != Block.STONE.blockID) {
             return false;
         }
         int i6 = 0;
-        if (world.getBlockId(x - 1, y, z) == Block.stone.blockID) {
+        if (world.getBlockId(x - 1, y, z) == Block.STONE.blockID) {
             ++i6;
         }
-        if (world.getBlockId(x + 1, y, z) == Block.stone.blockID) {
+        if (world.getBlockId(x + 1, y, z) == Block.STONE.blockID) {
             ++i6;
         }
-        if (world.getBlockId(x, y, z - 1) == Block.stone.blockID) {
+        if (world.getBlockId(x, y, z - 1) == Block.STONE.blockID) {
             ++i6;
         }
-        if (world.getBlockId(x, y, z + 1) == Block.stone.blockID) {
+        if (world.getBlockId(x, y, z + 1) == Block.STONE.blockID) {
             ++i6;
         }
